azure/ConsiderationsForServiceDesign.md

```diff
@@ -84,6 +84,74 @@ As you build out your service and API, there are a number of decisions that can
 
 :ballot_box_with_check: **YOU SHOULD** implement [conditional requests](https://tools.ietf.org/html/rfc7232) early. This allows you to support concurrency, which tends to be a concern later on.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a great developer experience.
+
+Resources are described by nouns. Resource and property names must be descriptive and easy for customers to understand.
+Avoid names that refer to service implementation details, e.g. "TreeLeafNode", or simply describe the structure of value without
+describing its purpose, e.g. "KeyValuePair".
+Ease of understanding comes from familiarity and recognition; you should favor consistency with other Azure services, names in the product's portal/user interface, and industry standards.
+
+Names should aid developers in discovering functionality without having to constantly refer to documentation.
+Use common patterns and standard conventions to aid developers in correctly guessing common property names and meanings.
+Use verbose naming patterns and avoid abbreviations other than
+well-known acronyms in your service domain.
+
+:heavy_check_mark: **DO** use the same name for the same concept and different names for different concepts wherever possible.
+
+### Recommended Naming Conventions
```

Comment:
How about the following:
"DO NOT use HTTP constructs in names (FooResult, FooRequest)".

Reply:
We originally included this guidance but removed it as it mainly pertains to schema names which are not part of the formal API contract.

But we should include this in a update to the OpenAPI Style Guide.